contentcuration/search/viewsets/contentnode.py

```diff
@@ -54,168 +50,127 @@ def filter_channel_list(self, queryset, name, value):
         user = not self.request.user.is_anonymous and self.request.user
         channel_ids = []
         if value == "public":
-            channel_ids = Channel.objects.filter(public=True, deleted=False).values_list("id", flat=True)
+            channel_ids = Channel.objects.filter(public=True, deleted=False, main_tree__published=True).values_list("id", flat=True)
         elif value == "edit" and user:
-            channel_ids = user.editable_channels.values_list("id", flat=True)
+            channel_ids = user.editable_channels.filter(main_tree__published=True).values_list("id", flat=True)
         elif value == "bookmark" and user:
-            channel_ids = user.bookmarked_channels.values_list("id", flat=True)
+            channel_ids = user.bookmarked_channels.filter(main_tree__published=True).values_list("id", flat=True)
         elif value == "view" and user:
-            channel_ids = user.view_only_channels.values_list("id", flat=True)
+            channel_ids = user.view_only_channels.filter(main_tree__published=True).values_list("id", flat=True)
         return queryset.filter(channel_id__in=list(channel_ids))
 
     def filter_keywords(self, queryset, name, value):
-        filter_query = Q(title__icontains=value) | Q(description__icontains=value)
-        tags_node_ids = ContentNode.tags.through.objects.filter(
-            contenttag__tag_name__icontains=value
-        ).values_list("contentnode_id", flat=True)[:250]
-        # Check if we have a Kolibri node id or ids and add them to the search if so.
-        # Add to, rather than replace, the filters so that we never misinterpret a search term as a UUID.
-        # node_ids = uuid_re.findall(value) + list(tags_node_ids)
-        node_ids = uuid_re.findall(value)
-        for node_id in node_ids:
-            # check for the major ID types
-            filter_query |= Q(node_id=node_id)
-            filter_query |= Q(content_id=node_id)
-            filter_query |= Q(id=node_id)
-        for node_id in tags_node_ids:
-            filter_query |= Q(id=node_id)
-
-        return queryset.filter(filter_query)
+        return queryset.filter(Q(keywords_tsvector=get_fts_search_query(value))
+                               | Q(author_tsvector=get_fts_search_query(value)))
```

Comment:
Why is the `author_tsvector` included here?

Reply:
To let users search nodes by author as possible keyword, am fine if we don't want it there